lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -5,6 +5,9 @@ module RubyIndexer
   class IndexVisitor < YARP::Visitor
     extend T::Sig
 
+    IGNORED_MAGIC_COMMENTS =
+      /^\s*#\s*(frozen_string_literal|typed|compiled|encoding|shareable_constant_value|warn_indent|rubocop|nodoc|doc):/i
```

Comment:
You're missing `coding` and `warn_past_scope` from [this list](https://github.com/ruby/ruby/blob/v3_0_0/parse.y#L8085-L8094).

A lot of the magic comments defined here are non-standard ones. I wonder if we could just ignore all comments attached to the top of the file and on the form `# <key>:<value>` rather than maintaining a list.

Reply:
> It would also be nice to not show licenses written in comments on top of files. They are usually huge and bury the actual documentation way down.

Don't we already break on empty lines when we collect the comments associated to an entity? If the license is properly separated from the comment of the first entity it should be okay:

```rb
# typed: true
# 
# Some license

# First comment
class Foo; end
```

For the `:nodoc:` etc, could we use the `index.yaml` and default values to make it more configurable?

Reply:
That's a great idea. I made it configurable in the `.index.yml` file.